include/directx/d3dx12.h

```diff
@@ -3293,6 +3293,7 @@ inline bool operator==( const D3D12_RENDER_PASS_DEPTH_STENCIL_DESC &a, const D3D
 #define D3DX12_COM_PTR_GET(x) x.p
 #define D3DX12_COM_PTR_ADDRESSOF(x) &x.p
 #endif
+#include <variant>

```

Comment:
That would cause a different public API surface depending on whether or not variant is included. That inconsistency seems less than ideal, right? Would it be better to just use a different approach in this case, such as returning an HRESULT and returning the FeatureSupport class via out param?

Reply:
That seems okay to me, since this is header-only. If you want to use a `std::variant`-based API, then do so. If you don't want the bloat required to do that, then opt out.

Reply:
Resolved using macros. I used `__cplusplus` and `_MSVC_LANG` for identifying the version of C++ standard, the latter of which may only work in Visual studio. The user also need to define a specific flag in order to use the <variant> option. If it's not turned on, the Create() factory method will return only the feature, not the HRESULT.